htmresearch/algorithms/apical_dependent_temporal_memory.py

```diff
@@ -174,34 +182,49 @@ def compute(self,
     Whether to grow / reinforce / punish synapses
     """
 
+    self.burstingColumns = []
     if basalGrowthCandidates is None:
       basalGrowthCandidates = basalInput
 
     if apicalGrowthCandidates is None:
       apicalGrowthCandidates = apicalInput
 
     # Calculate predictions for this timestep
-    (activeBasalSegments,
-     matchingBasalSegments,
-     basalPotentialOverlaps) = self._calculateSegmentActivity(
-       self.basalConnections, basalInput, self.connectedPermanence,
-       self.activationThreshold, self.minThreshold)
-
     (activeApicalSegments,
-     matchingApicalSegments,
-     apicalPotentialOverlaps) = self._calculateSegmentActivity(
-       self.apicalConnections, apicalInput, self.connectedPermanence,
-       self.activationThreshold, self.minThreshold)
-
-    predictedCells = np.intersect1d(
-      self.basalConnections.mapSegmentsToCells(activeBasalSegments),
-      self.apicalConnections.mapSegmentsToCells(activeApicalSegments))
+      matchingApicalSegments,
+      apicalPotentialOverlaps) = self._calculateSegmentActivity(
+        self.apicalConnections, apicalInput, self.connectedPermanence,
+        self.activationThreshold, self.minThreshold)
+
+    apicallySupportedCells = self.apicalConnections.mapSegmentsToCells(
+        activeApicalSegments)
+    if self.useFeedback:
```

Comment:
I think it'd be reasonable to remove this `self.useFeedback` check and just use `_calculateBasalSegmentActivity`. It already checks whether `reducedBasalThreshold != activationThreshold`, so there's no real performance overhead.

So we can remove `self.useFeedback` and remove the `else` statement.